### PR TITLE
Add PreventIncompatibleGames to ClientConfiguration

### PR DESCRIPTION
**Pull request by devo1929**

### CODE REVIEW

File: DXMainClient/DXGUI/Multiplayer/GameFiltersPanel.cs [hunk 1]
```diff
@@ -70,11 +70,13 @@ public override void Initialize()
                 gap, chkBoxHidePasswordedGames.Y + UIDesignConstants.BUTTON_HEIGHT + gap,
                 0, 0
             );
+            chkBoxHideIncompatibleGames.Visible = !ClientConfiguration.Instance.PreventIncompatibleGames;
```
Comment (devo1929) [inline DXMainClient/DXGUI/Multiplayer/GameFiltersPanel.cs:73]: If the client def is preventing incompatible games, then this check box is not necessary.

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs [hunk 2]
```diff
@@ -1489,6 +1489,10 @@ private void GameBroadcastChannel_CTCPReceived(object sender, ChannelCTCPEventAr
                 if (cncnetGame == null)
                     return;
 
+                bool incompatible = cncnetGame == localGame && gameVersion != ProgramConstants.GAME_VERSION;
+                if (incompatible && ClientConfiguration.Instance.PreventIncompatibleGames)
+                    return;
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:1494]: IMO would be better to do this at join level, as in add a message or a messagebox stating that this game is incompatible and you can't join it.

**Review by Metadorius — CHANGES_REQUESTED** [review]

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs (hunk 2 above)
Comment (devo1929) [reply DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:1494]: I'm ok with that. We already have error messages that appear for other reasons. I'll add this one.

Comment (devo1929) [reply DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs]: Done.

File: ClientCore/ClientConfiguration.cs [hunk 3]
```diff
@@ -330,6 +330,12 @@ public IEnumerable<string> SupplementalMapFileExtensions
             => clientDefinitionsIni.GetStringValue(SETTINGS, "SupplementalMapFileExtensions", null)?
                 .Split(',', StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
 
+        /// <summary>
+        /// This prevents users from joining games that are incompatible/on a different game version than the current user.
+        /// Default: false
+        /// </summary>
+        public bool DisallowJoiningIncompatibleGames => clientDefinitionsIni.GetBooleanValue(SETTINGS, "DisallowJoiningIncompatibleGames", false);
```
Comment (Rans4ckeR) [inline ClientCore/ClientConfiguration.cs:337]:
```suggestion
        public bool DisallowJoiningIncompatibleGames => clientDefinitionsIni.GetBooleanValue(SETTINGS, nameof(DisallowJoiningIncompatibleGames), false);
```

**Review by Rans4ckeR — APPROVED** [review]

**Pull request merged**